### ISSUE TITLE
Installing QNAP version does not work.

### ISSUE THREAD

**Issue by toxic17** [SickRage/sickrage-issues#355]
OK Running QTS 4.2.1 1224 on a TVS-463

Installed SickBeard-TVRage_151227.qpkg (python 2.7.9 and GIT 2.1.0 installed)

once installed opening Sickbeard-RageTV I get 

No data received

ERR_EMPTY_RESPONSE

in Browser.

when i check /share/CACHEDEV1_DATA/.qpkg/SickBeard-TVRage folder for contents they are only 13 files and 8 folders with the combined file size of 1,472Kb

is this what is mean to be in this folder, it seems very light.

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by huush** [comment]
Hi @toxic17,

I just installed that version to switch to this repo about an hour ago. Had a few teething issues, but basically yes. That is what that will do if you aren't bringing across an existing db. 

After the install is completed, start SickRage in the QNAP admin at http://yourserver:8080

Go to App Center and turn SickRage on. Then select OPEN. This will open up a basic web page asking you what repo you want to install from, 1. The original, 2. This one (but a http version) or 3. A custom. 

I found that selecting option 2 wasn't working for some reason. 

So, I copied the https version URL  https://github.com/SickRage/SickRage.git into the custom field. This also failed a couple of times. What seemed to work was after pasting the URL hit the SPACE bar and the field should turn yellow. Remove the space for good measure and the field should still be yellow. It seems only if it goes yellow does then selecting the button beneath the input field work.

After you press the button, the repo should be installed quite quickly. The code is very light. It's the db and image caching that starts taking up all the space. So, if it's a fresh install and you have no previous dbs and cache then that's it. http://yourserver:7073 should present the main page.

If you have backed up a previous version, stop sickrage again. Then copy the db and config ini files back into the relevant directories - and possibly the config/cache directories containing all the images and thumbnails to save downloading these again. @Fetchered seems to have a similar QNAP setup to you in https://github.com/SickRage/sickrage-issues/issues/3 Just search for QNAP in that thread for where files go.

Then restart.

**Comment by labrys** [comment]
@clinton-hall Any idea why option 2 didn't work for @huush?

**Comment by clinton-hall** [comment]
Not sure, works fine here.
Depends on what actually didn't work... To be clear, this should simply save a file with the URL to the chosen repo (saved in config, so after app upgrade you won't be asked again).
This then loads a simple webpage saying to restart the app (and the web server then dies).
On restart of app, the text file is read to get the actual url, then git does its thing... While http URLs is entered, the script tries do clone with git:// first...

So basically, I'm curious to know if the text file was created for option 2, if so, was it just failing to load the page asking for app restart?

**Comment by toxic17** [comment]
labrys closed this 4 hours ago???

how is this closed when i had not even answered?

now

>After the install is completed, start SickRage in the QNAP admin at http://yourserver:8080
>Go to App Center and turn SickRage on. Then select OPEN. This will open up a basic web page >asking you what repo you want to install from, 1. The original, 2. This one (but a http version) or 3. >A custom.

if you care to read my initial post, this does not work, full stop.  Sickrage is enabled, installed, but when i goto the WEB URL all is get is this:

![capture](https://cloud.githubusercontent.com/assets/8934300/12032214/b66de148-ae0d-11e5-9637-b5436c6d26be.JPG)

do you understand?

again?

![capture1](https://cloud.githubusercontent.com/assets/8934300/12032229/dcbdc070-ae0d-11e5-953b-7941fddd5600.JPG)

there is no webpage to setup Sickrage. i checked the config.ini file it is using 7073 port. still there is no webpage loading.

**Comment by labrys** [comment]
@toxic17 This was closed as the issue you are having is with the QNAP installation itself, prior to even installing SickRage, thus the label UPSTREAM.

**Comment by labrys** [comment]
@toxic17 There is no reason to be rude about the issue.  Within less than 15 minutes you went from posting a rude response to changing the title to a demanding 'PLEASE OPEN!!!' to posting a new issue all while I was trying to respond to your first response.  A polite statement that your issue was not yet resolved would have sufficed.

**Comment by clinton-hall** [comment]
is this related to #297 ?
Perhaps the version of python doesn't include the SimpleHTTPServer module...

**Comment by huush** [comment]
Attitude @toxic17 . Seriously. I "cared enough to read your initial post" and what I responded to is what I took away from your limited information.

Should have mentioned @clinton-hall, when the button was pressed the url#restart (can't remember exactly the URL that came up) failed to load with a similar, or the same error as @toxic17 is reporting.  There's no mention of the URL he's attempting to load and has cut it from his screenshots so it still seems possible. It's also unclear whether he's attempting a new install or upgrade (which is my case).

Not sure why things weren't working for me. It was dumb luck that I pasted the git command line into the input field rather than just the URL and so had to edit the field and saw it become yellow. I'm using Win 7 and latest Chrome. Perhaps the input variable was not properly initialised and only detected after making a manual change. Seems unlikely, but that's the net effect of what I experienced. I had tried manually restarting the app about 4 times to no effect before having that success.

**Comment by dltyih** [comment]
@clinton-hall, as far as I am concerned, yes it is related to #297 in some ways because my original issue (in #297) was brought about by not being able to get SR functioning on the QNAP using the QTS supplied Python 2.7.3, and I was not sophisticated enough in Python and ssh to understand how to get SR to reference the more up to date QPython. 

I did try your latest package here: https://www.dropbox.com/s/j1svazqdi9ieq82/SickBeard-TVRage_151227.qpkg but for whatever reason that was out of my depth, I could not get it to function beyond seeing the selection page of which repo to install.

I have, however, noticed another thread in QNAP forum just put up by mannibis http://forum.qnap.com/viewtopic.php?f=223&t=118366&p=525858#p525858 which offers a way to install Entware Python 2.7.10 and followed by your aforementioned SR package. This seemed to have worked for me (I say "seemed" because I am still testing, but so far so good). So for the time being and until I can find a cleaner solution which does not involve Entware, I am gradually moving my setup under that. So far I did SR and SABnzbd+. Tomorrow I will try CouchPotato, Mylar and Headphones. I will try to see if I can also remove the QTS supplied Python 2.7.3.

Happy New Year, everyone!

**Comment by toxic17** [comment]
I can confirm this is now working when using Python 2.7.10 (entware)  I did try the QPython2 (1Gb in size!!!) which is 2.7.11 but could not get it to work.

Clinton, you need to update your initial post here as it is so outdated and misleading now.

http://forum.qnap.com/viewtopic.php?f=223&t=35015#p152121

**Comment by clinton-hall** [comment]
Yes, I do need to update the forum posts, but I am waiting until I have a confirmed working app and a list of dependencies.

**Comment by miigotu** [comment]
@clinton-hall those import errors before aren't even related to sr at all. None of SR dependencies require pastescript for example. the python package that had that error was simply broken.
